app_vue/src/components/sensorMap.vue

```diff
@@ -1,35 +1,68 @@
 <script setup>
-  import { reactive, ref, onMounted, watch } from 'vue'
+  import { reactive, onMounted, watch, onBeforeMount, onDeactivated } from 'vue'
   import SensorMapMarker from '@/components/sensorMapMarker.vue';
   import 'leaflet/dist/leaflet.css'
-  import { LMap, LTileLayer, LMarker, LControlZoom, LPolyline } from '@vue-leaflet/vue-leaflet'
+  import { LMap, LTileLayer, LMarker, LControlZoom, LPolyline, LIcon } from '@vue-leaflet/vue-leaflet'
   import { useDevice, DEVICE_SIZE } from '@/utils/screenSizeHelper';
+  import { getLatLng } from '@/utils/getLatLngFromAddressHelper';
   import { useSensorStore } from '@/stores/sensors_store';
   import { useRouteStore } from '@/stores/route_store';
   import { storeToRefs } from 'pinia';
 
-  const center = ref([43.7, -79.42]); // TODO: update to possibly be user's current location
-  const zoom = ref(10);
+  // stores
   const sensorStore = useSensorStore();
   const { sensors } = storeToRefs(sensorStore);
+  const routeStore = useRouteStore();
+  const { 
+    getSelectedRouteLatLong, 
+    getIsRouteOptimized } = storeToRefs(routeStore);
+
   const state = reactive({
     location: 'bottomright',
-    device: useDevice()
+    device: useDevice(),
+    isRouteOptimized: false,
+    polyLineLatLngs: [],
+    startPointLatLng: [],
+    endPointLatLng: [],
+    zoom: 10,
+    center: [43.7, -79.42]
   });
-  const polyLineLatLngs = ref([]);
 
-  const routeStore = useRouteStore();
-  const { getSensorRouteLatLong } = storeToRefs(routeStore);
+  const startImgUrl = 'src/assets/images/feather-disc.svg';
+  const endImgUrl = 'src/assets/images/feather-map-pin.svg';
 
-  watch(getSensorRouteLatLong, () => {
-    polyLineLatLngs.value = routeStore.getSensorRouteLatLong;
-  }, { deep: true })
+  onBeforeMount(async () => {
+    // processing start/end/center lat lng points
+    state.startPointLatLng = await getLatLng(routeStore.getStartPointAddress);
+    state.endPointLatLng = await getLatLng(routeStore.getEndPointAddress);
+    state.center = state.startPointLatLng;
+
+    routeStore.setHasMappedStartEnd(true);
+  })
```

Comment:
gets start and end point lat lng and sets the center of the map to be the start point